pkg/optin/optin.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2021 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package optin
+
+import (
+	"regexp"
+	"sort"
+	"strings"
+
+	generator "k8s.io/kube-state-metrics/v2/pkg/metric_generator"
+)
+
+// MetricFamilyFilter filters metric families which are defined as opt-in by their generator.FamilyGenerator
+type MetricFamilyFilter struct {
+	metrics []*regexp.Regexp
+}
+
+// Test tests if a given generator is an opt-in metric family and was passed as an opt-in metric family at startup
+func (filter MetricFamilyFilter) Test(generator generator.FamilyGenerator) bool {
+	if generator.OptIn {
```

Comment:
Can we reduce the nesting with
```
  if !generator.OptIn {
    return true
  }
```

Reply:
Will apply this change, agree that the nesting is a bit too much but didn't think about an early-return like you are suggesting. Thanks! 🎉